Fix/logging user action

katedmc opened the pull request:
# Description
<!-- Summarize your PR here. Think "what, why, how." Bonus points for including an estimate of how long this PR will take to read, review, and test. -->
What:
I extended the existing logging code to include all authoring app actions. The logging file appends user actions and has a max file size.

Why:
A previous issue asked for detailed logging. We now have more detailed information about what actions the users take.

How:
Utilized a different constructor for the FileHandler to add appending to the file, and changed the file limit from default to 1MB
Leveraged the logger from the GUI in InsertBefore and RightPanel classes to add logging. 

Time:
Less than 15 minutes to read and review the code and PR.
Approximately 15 minutes to test code changes. 

## Added
<!-- What features, code, or content did you add in this PR, if any? For this section as well as the Changed/Removed sections, use bullet points to stay concise. Remove any sections that you leave empty. -->
- Added a constant value for the max log file size in GUI class.
- Added boolean value "true" to FileHandler constructor for appending
- Added logger lines into InsertBefore and RightPanel Class 

## Changed

- Changed to different FileHandler constructor to use file appending
- Changed max file value from default max to 1MB with constant value


# How should this be manually tested?

Checking Max File Size:
1. Change 'fileSizeLimit" to 200
2. Run Authoring app
3. Select "New Scenario"
4. Select "New Item"
5. Check log for added items
6. Select "New Item" x2
7. Check if log file has been reset.
8. Change "fileSizeLimit back to "1024000"

Checking All user actions are logged:
1. Run authoring app
2. Select "New Scenario"
3. Select all available user options from "New Item" to "Test Scenario"
4. Check if log file contains selected user actions

katedmc committed:
Added in logging of all user actions.

Extended the logging to catch all user actions. Log file is now appended
to.

katedmc committed:
Added 1MB size limit for log file.

Added in size limit for logger now that user actions are appended to
file.

danielmapar commented on src/authoring/GUI.java:
```diff
@@ -47,6 +47,7 @@
  */
 public class GUI extends JFrame {
 	private static final long serialVersionUID = -1291725446662111704L;
+	private static final int fileSizeLimit = 1024000; //1MB file size limit for log file
```
Maybe we could pump the limit to 100mb, or only enable logging while on debug mode. Right now logging will be generated regardless if we are debugging the app or not, that means the production version of it (the one being used in the wild) will be generating logs all day. The app would end up breaking after some usage

katedmc replied:
The logger class now has a file limit of 100mb per log file.

katedmc committed:
TBB Logger Class

This class will be used consolidate logging code across TBB.

katedmc committed:
GUI Logger changed to TBBLogger

The GUI logger is now the TBBLogger, and all classes that use the GUI
logger are included in the new logging design.

katedmc committed:
Player Logger changed to TBBLogger

Abstract Player class, and subsequent child classes, now all use
TBBLogger to help consolidate code usage.

katedmc committed:
ScenarioParser changed to use TBBLogger

The logger has been changed to use the TBB logger to clean up the code
more. The ScenarioParser also uses the static error logger from
TBBLogger to further clean up code.

katedmc committed:
Logging added to SCALP

TBBLogger added to SCALP and logging added to capture Scenario
Selection.

katedmc committed:
Merge branch 'develop' into fix/logging_user_action

katedmc commented:
As of now, on top of the original changes, this branch:

- has the new TBBLogger class for better code reuse;
- all old instances of Logger have been replaced by TBBLogger
- All repeated uses of the FileHandler format have been removed.
- SCALP logging has been to this branch (A 2 line change in SCALP)

The overall functionality hasn't been changed as the main changes were to the logger instances.
